localstack/config.py

```diff
@@ -83,10 +85,73 @@ def __init__(
         self.logs = logs
 
     @staticmethod
-    def from_config():
+    def defaults() -> "Directories":
+        """Returns Localstack directory paths based on the localstack filesystem hierarchy."""
+        return Directories(
+            static_libs="/usr/lib/localstack",
+            var_libs=f"{DEFAULT_VOLUME_DIR}/lib",
+            cache=f"{DEFAULT_VOLUME_DIR}/cache",
+            tmp=f"{DEFAULT_VOLUME_DIR}/tmp",
+            functions=f"{DEFAULT_VOLUME_DIR}/tmp",  # FIXME: remove - this was misconceived
+            data=f"{DEFAULT_VOLUME_DIR}/data",
+            logs=f"{DEFAULT_VOLUME_DIR}/logs",
+            config="/etc/localstack/conf.d",  # for future use
```

Comment:
What exactly will the future use of the config directory be? Currently, we use it to define config profiles which are only used by the CLI. With this change, the config directly in the container will be empty (since nothing will mount to  `/etc/localstack` for now).

Reply:
fair point. i think the confusion in general comes from the conflation of a) CLI config with b) runtime config in container mode, and c) runtime config in host mode. part of what this PR addresses is the clearer separation of those. `CONFIG_DIR` is purely a CLI option that holds configuration for the CLI (which in turn again can manipulate localstack config). however, there's a case for using custom configurations without the CLI.